index_trade.html

```diff
@@ -0,0 +1,263 @@
+      .range([height, 0])    
+      .nice();
+      
+    // AXES
+      var xAxis = d3.svg.axis()
+      .scale(x)
+      .tickSize(-height);
+      
+      var yAxis = d3.svg.axis()
+      .scale(y)
+      .tickSize(-width)
+      .orient('left');   
+      
+      svg.append('g')
+          .attr('class', 'x axis')
+          .call(xAxis)
+          .attr('transform', 'translate(0,' + (height) + ')')
+        .append('text')
+          .attr('x', width/2)
+          .attr('y', 70)
+          .style('text-anchor', 'middle')
+          .text('Trade Balance Between US and UK');      
+      
+      svg.append('g')
+          .attr('class', 'y axis')
+          .call(yAxis)
+        .append('text')
+          .attr('transform', 'rotate(-90)')
+          .attr('y', -60)
+          .attr('x', -height/2)      
+          .style('text-anchor', 'middle')      
+          .text('Exchange Rate, GBP/USD');           
+      
+      // GRID LINES
+      
+      var ygrid = svg.append('g')
+      .attr('class', 'y axis')
+      .call(yAxis);
+      
+      ygrid.selectAll('g').filter(function(d) { return d; })
+        .classed('minor', true);
+       
+    // EVENT HANDLERS
+    d3.select('#year-range').on('change', function () {
+      options.yearRange = d3.event.target.value;
+      update();
+    });
+  }
+    
+  function update() {
+    txData = data.slice();
+    if (options.yearRange !== 'all') {
+      var yearRange = options.yearRange.split('-');
+      txData = txData.filter(function (d) {
+        return d.year >= yearRange[0] && d.year <= yearRange[1];
+      });
+    }
+      
+    // SCATTERPLOT
+    var points = svg.selectAll('.point')
+      .data(txData);
+    points.enter().append('circle')
+      .attr('class', 'point')
+      .attr('r', 5);
+    points
+      .attr('cx', function (d) { return x(d.trade); })
+      .attr('cy', function (d) { return y(d.rate); });
+    points.exit().remove();
+
+      // HISTOGRAM
+      
+      // x hist
+      
+    var xmap = data.map(function(d) { return d.trade; })
+    var xhist = d3.layout.histogram().bins(x.ticks())(xmap);      
+      
+      console.log('hist', xhist);
+      
+      bwidth = x(x.ticks()[1])/2;
+      
+    var bars = svg.selectAll('.bar')
+        .data(xhist)
+        .enter()
+        .append('g')
+        .attr('class', 'bar')
+        .attr('transform', function(d, i) { return 'translate(' +  
+              ( x(x.ticks()[i+1]) - (x(x.ticks()[1]))/2 - (bwidth/2) ) + ')'; });
+          
+    bars.append('rect')
+        .attr('x', 0)
+        .attr('y', function(d) { return height-(d.y); })
+        .attr('width', bwidth)
+        .attr('height', function(d) { return d.y; }); 
+  
+      // y hist
+      
+    var ymap = data.map(function(d) { return d.rate; })
+    var yhist = d3.layout.histogram().bins(y.ticks())(ymap);      
+      
+      console.log('rate', yhist);
+      
+    var bars = svg.selectAll('.bar')
```

Comment:
Not only are we overwriting the `bars` variable, but our selection of anything classed `.bar` means that we've selected all of the bars we created above for the y axis. Lucky for us, we have one more bar in the Y axis histogram than we do in the X axis histogram, so the enter selection contains a solitary data point for us to append. This is why we only have the one bar on the Y axis.